### PR TITLE
Lazy download of data splits 

### PR DESCRIPTION
**Pull request by Neeratyoy**
#### What does this PR implement/fix? Explain your changes.
Addresses the pending part from #346 . Wherein, the splits and the data are not downloaded when `get_task` is called. Only when a run is executed, the data and the respective splits are downloaded.
A unit test has been added for the lazy fetch of a task.

#### How should this PR be tested?
Calling `get_task` or `get_tasks` with the additional parameter of `download_data=False` downloads only the task.xml file. Which can be verified by monitoring the cache directory. When a task is run, the corresponding datasplits.arff file is downloaded and available in the cache directory.

#### Any other comments?
- Earlier: `task.class_labels` assignment was happening in a nested manner after `task.download_split`
- Now: `task.class_labels`is being assigned even if splits are not being downloaded, by fetching it from dataset's description

### CODE REVIEW

File: tests/test_tasks/test_task_functions.py [hunk 1]
```diff
@@ -129,6 +129,24 @@ def test_get_task(self):
...
+        task = openml.tasks.get_task(2, download_data=False)
+        self.assertIsInstance(task, OpenMLTask)
+        self.assertTrue(os.path.exists(os.path.join(
+            self.workdir, 'org', 'openml', 'test', "tasks", "2", "task.xml",
+        )))
+
+        self.assertFalse(os.path.exists(os.path.join(
+            self.workdir, 'org', 'openml', 'test', "tasks", "2", "datasplits.arff"
+        )))
+        # Not checking datasets/ as other test cases might interfere
```
Comment (mfeurer) [inline tests/test_tasks/test_task_functions.py]: The cache directories are unique for each unit test so this should not be an issue: https://github.com/openml/openml-python/blob/master/openml/testing.py#L59

File: tests/test_tasks/test_task_functions.py [hunk 2]
```diff
@@ -129,6 +129,24 @@ def test_get_task(self):
             self.workdir, 'org', 'openml', 'test', "datasets", "1", "dataset.arff"
         )))
 
+    def test_get_task_lazy(self):
+        task = openml.tasks.get_task(2, download_data=False)
+        self.assertIsInstance(task, OpenMLTask)
```
Comment (mfeurer) [inline tests/test_tasks/test_task_functions.py]: Could you please add a check that the class labels are retrieved despite the data not downloaded?

File: openml/tasks/functions.py [hunk 3]
```diff
@@ -275,31 +275,43 @@ def __list_tasks(api_call):
...
     for task_id in task_ids:
-        tasks.append(get_task(task_id))
+        tasks.append(get_task(task_id, download_data))
     return tasks
 
 
-def get_task(task_id):
-    """Download the OpenML task for a given task ID.
+def get_task(task_id, download_data=True):
+    """Download the OpenML task representation for a given task ID, optionally
```
Comment (mfeurer) [inline openml/tasks/functions.py]: Could you please try to keep the first line of the docstring to be a single line and add information about optional arguments afterwards?

File: openml/tasks/functions.py [hunk 4]
```diff
@@ -275,31 +275,45 @@ def __list_tasks(api_call):
...
 
 
-def get_task(task_id):
-    """Download the OpenML task for a given task ID.
+def get_task(task_id, download_data=True):
+    """Download task representation for a given task ID
     Parameters
     ----------
-    task_id : int
+    The task representation is downloaded while the download of data
```
Comment (mfeurer) [inline openml/tasks/functions.py]: Sorry, that's not how I meant it. Could you please check for example the doc strings of the [scikit-learn random forest](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/forest.py#L750)? There's a single line, then some additional information (where I think the sentence about the optional download of the split and data should go) and then the `Parameters`.

Comment (Neeratyoy) [reply openml/tasks/functions.py]: sorry about that, should have looked up an example, will make the changes
